[RHACS] [Docs] [4.5 and later versions] ROX-21863: Setting up image mirroring for registries (#82567)
openshift/openshift-docs: modules/scanning-images-by-using-secured-clusters.adoc

// Module included in the following assemblies:
//
// * operating/examine-images-for-vulnerabilities.adoc

:_mod-docs-content-type: PROCEDURE
[id="scanning-images-by-using-secured-clusters_{context}"]
= Scanning images by using secured clusters

To scan images by using the secured clusters instead of the Central services, you can use the delegated image scanning feature.

> A new delegated scanning configuration specifies the registries from which you can delegate image scans. For images that Sensor observes, you can use the delegated registry configuration to delegate scans from no registries, all registries, or specific registries.

Reviewer: I won't hold up merging but for the future, we try to avoid using "new" in anything except release notes, since after a few releases, it won't be new.

To enable delegation of scans by using the `roxctl` CLI, Jenkins plugin, or API, you must also specify a destination cluster and source registry.

.Prerequisites

* You have installed Scanner in the secured cluster to scan images.
+
[NOTE]
====

> Enabling Scanner-slim is supported on {ocp} and Kubernetes secured clusters.

Reviewer 2: nit: 
```suggestion
Enabling Scanner is supported on {ocp} and Kubernetes secured clusters.
```
Or call out both "Scanner-slim" and "Scanner V4 Indexer"

====

.Procedure

. In the {product-title-short} portal, click *Platform Configuration -> Clusters*.
. In the *Clusters* view header, click *Delegated scanning*.
. In the *Delegated Image Scanning* page, provide the following information:

** *Delegate scanning for*: To choose the scope of the image delegation, select one of the following options:

> *** *None*: The default option. This option specifies that the secured clusters do _not_ scan any images, except for images from the {ocp} integrated registry.

Reviewer 2: nit: recognize this was just copied, but based on [the comment here](https://github.com/openshift/openshift-docs/pull/80974#discussion_r1747177617), the preferred way to refer to the integrated registry seems to be:

```suggestion
*** *None*: The default option. This option specifies that the secured clusters do _not_ scan any images, except for images from the integrated OpenShift image registry.
```

// Module included in the following assemblies:
//
// * operating/examine-images-for-vulnerabilities.adoc

:_mod-docs-content-type: PROCEDURE
[id="scanning-images-by-using-secured-clusters_{context}"]
= Scanning images by using secured clusters

> By configuring delegated image scanning, you can scan images by using the secured clusters instead of the Central services.

Reviewer 3: Consider "You can scan images by using secured clusters instead of Central services by configuring delegated image scanning."

> A new delegated registry configuration specifies the registries from which you can delegate image scans. For images that Sensor observes, you can use the delegated registry configuration to delegate scans from no registries, all registries, or specific registries. 

Reviewer 3: I'm not that familiar with this part of the docs. I'm assuming that capitalized things in running text like "Sensor" and "Scanner-slim" are proper product names.

Author: Those are the major components of ACS

> A new delegated registry configuration specifies the registries from which you can delegate image scans. For images that Sensor observes, you can use the delegated registry configuration to delegate scans from no registries, all registries, or specific registries. 

Reviewer 3: "Newly delegated" if "delegated" is a descriptor of yours. "New delegated" is fine if the feature name is "delegated registry configuration."

> A new delegated registry configuration specifies the registries from which you can delegate image scans. For images that Sensor observes, you can use the delegated registry configuration to delegate scans from no registries, all registries, or specific registries. 

Reviewer 2: This may be relic from before we had the UI available, I would recommend referring to it as the `delegated scanning configuration` instead of `delegated registry configuration` to more closely resemble the UI:

![image](https://github.com/user-attachments/assets/7a49b4ef-29b8-4261-876b-ebbb99e606eb)

To enable delegation of scans by using the `roxctl` CLI, Jenkins plugin, or API, you must also specify a destination cluster and source registry.

.Prerequisites

> * You have installed Scanner-slim in the secured cluster to scan images.

Reviewer 2: With the inclusion of Scanner V4, perhaps references to `scanner-slim` in this section can be changed to just `scanner` or callout both Scanner-slim and Scanner V4 Indexer (refer to architecture page for more details)

+
[NOTE]
====
Enabling Scanner-slim is supported on {ocp} and Kubernetes secured clusters.
====

* You have set up image mirroring for registries.
+

> For more information, see "Enabling seamless image access and efficient use of multiple registry sources".

Reviewer 2: Setting up image mirroring is not a pre-req for using delegated scanning, the two are not mutually exclusive. Recommend removing this line item from the pre-reqs.

Delegated scanning will scan images via the secured cluster. IF the user happens to have mirroring configured, delegated scanning will automatically attempt to scan images from the mirror references, however mirroring is NOT required to be setup to use delegated scanning, and in most user environments mirroring is NOT setup.

.Procedure

. In the {product-title-short} portal, click *Platform Configuration -> Clusters*.
. In the *Clusters* view header, click *Delegated scanning*.
. In the *Delegated Image Scanning* page, provide the following information:

** *Delegate scanning for*: To choose the scope of the image delegation, select one of the following options:
*** *None*: The default option. This option specifies that the secured clusters do _not_ scan any images, except for images from the {ocp} integrated registry.
*** *All registries*: This option indicates that the secured clusters scan all the images.
*** *Specified registries*: This option specifies the images that secured clusters should scan based on the registries list.

> ** *Select default cluster to delegate to*: From the drop-down list, select the name of the default cluster. The default cluster processes the scan requests coming from the command-line interface (CLI) and API. This is optional and you can select `None` if required.

Reviewer 3: I don't know the GUI, but is it really necessary to explain these options here? Consider deleting much of this if there is appropriate contextual guidance.

Reviewer 2: These options have confused members of ACS and users alike, the extra details were added awhile back to help explain what the different settings mean (the verbosity is also in the API docs). I recommend keeping these details.

Side note: the GUI currently has inaccurate descriptions of how the configuration works (have ROX-23464 to address that)

> ** Optional: To specify the source registry and destination cluster details, click *Add registry*. 

Reviewer 3: Could this be raised up a level and made into an ordered list item?

Author: No, there is no particular sequence to do these steps. So I would prefer to keep it as an unordered list.

+
For example, specify the source registry as `example.com`, and select `remote` from the drop-down list for the destination cluster. You can add more than one source registry and destination cluster if required.
+
[IMPORTANT]
====
You can select the destination cluster as `None` if the scan requests are not coming from the CLI and API. 
====

. Click *Save*.

> Image integrations are now synchronized between Central and Sensor, and Sensor captures pull-secrets from each namespace. Sensor then uses these credentials to authenticate to the image registries.

Reviewer 3: ```suggestion
Image integrations are now synchronized between Central and Sensor, and Sensor captures pull secrets from each namespace. Sensor then uses these credentials to authenticate to the image registries.
```
